octoprint_mrbeam/util/material_csv_parser.py

```diff
@@ -3,21 +3,46 @@
 
 MRBEAM = "Mr Beam II"
 MRB_DREAMCUT = "MrB II Dreamcut"
+MRB_DREAMCUT_S = "MrB II Dreamcut S"
+
 MRB_READY = "MrB II Dreamcut Ready"  # not used yet
-MRB_DREAMCUT_NOT_VALIDATED = "Dreamcut (not validated)"
+MRB_DREAMCUT_NOT_VALIDATED = "Dreamcut (not validated)"  # not used yet
+MRB_DREAMCUT_S_NOT_VALIDATED = "Dreamcut S (not validated)"  # not used yet
 
-DEFAULT_LASER = MRBEAM
+DEFAULT_LASER_MODEL = "0"
+LASER_MODEL_S = "S"
 
 
-def model_id_to_csv_name(id):
+def model_ids_to_csv_name(device_model_id, laser_model_id):
     convert = {
-        octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2: MRBEAM,
-        octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2_DC: MRB_DREAMCUT,
-        octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2_DC_R1: MRBEAM,
-        octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2_DC_R2: MRBEAM,
+        (octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2, DEFAULT_LASER_MODEL): MRBEAM,
+        (
+            octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2_DC,
+            DEFAULT_LASER_MODEL,
+        ): MRB_DREAMCUT,
+        (
+            octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2_DC,
+            LASER_MODEL_S,
+        ): MRB_DREAMCUT_S,
+        (
+            octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2_DC_S,
+            DEFAULT_LASER_MODEL,
+        ): MRB_DREAMCUT,
+        (
+            octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2_DC_S,
+            LASER_MODEL_S,
+        ): MRB_DREAMCUT_S,
+        (
+            octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2_DC_R1,
+            DEFAULT_LASER_MODEL,
+        ): MRBEAM,
+        (
+            octoprint_mrbeam.util.device_info.MODEL_MRBEAM_2_DC_R2,
+            DEFAULT_LASER_MODEL,
+        ): MRBEAM,
```

Comment:
```py
from __future__ import absolute_imports

from . import device_info

# now use device_info.MODEL_MRBEAM2_DC_S
```

You can also disable the retarded black formatting with 
```py
# fmt: off

# write some unformatted code

# fmt: on
```

Just small bits :)